server/lib/referenceLookup.js

```diff
@@ -59,3 +78,31 @@ function indexIdentifiers() {
   });
 }
 
+// hack: handle special case of stock movements
+function stockMovementType(id) {
+  /**
+   * Stock Receipt API
+   * /receipts/stock/{{name}}/:document_uuid
+   *
+   * the {{name}} is what we define for example in { key : 'FROM_PURCHASE', path :'entry_purchase' },
+   *
+   * empty {{name}} means that there is no API entry for this name.
+   */
+  const flux = {
```

Comment:
Since this never changes, it would be easier to put it outside the function `stockMovementType`.  That way, the V8 engine will not allocate memory, remove it, allocate it, remove it, etc, on every HTTP request that hits this route.  If we make it a module-level constant, then it will be allocated at server start up and never freed.

```js
const flux = {
 // .... etc ...
};

function stockMovementType(id) {
  return flux[id];
}
```